src/realm/object-store/impl/realm_coordinator.cpp

```diff
@@ -102,6 +103,26 @@ void RealmCoordinator::create_sync_session()
         return;
 
     m_sync_session = m_config.sync_config->user->sync_manager()->get_session(m_db, *m_config.sync_config);
+    if (m_config.sync_config && m_config.sync_config->flx_sync_requested) {
+        std::weak_ptr<sync::SubscriptionStore> weak_sub_mgr(m_sync_session->get_flx_subscription_store());
+        sync::ClientReplication& history = static_cast<sync::ClientReplication&>(*m_db->get_replication());
+        history.set_write_validator_factory(
+            [weak_sub_mgr](Transaction& tr) -> util::UniqueFunction<sync::SyncReplication::WriteValidator> {
+                auto sub_mgr = weak_sub_mgr.lock();
+                if (!sub_mgr) {
+                    throw std::runtime_error("Subscription store was destroyed while user writes were on-going");
+                }
```

Comment:
It turns out we don't need to capture the sub_mgr at all. I think we still want to lock it and get the info out of it here so that we can cache the result across all the writes in this transaction.